common/common.go

```diff
@@ -314,3 +322,22 @@ func SystemdVersion(systemdBinaryPath string) (int, error) {
 
 	return version, nil
 }
+
+// FSSupportsOverlay checks whether the filesystem under which
+// a specified path resides is compatible with OverlayFS
+func FSSupportsOverlay(path string) bool {
+	var data syscall.Statfs_t
+	err := syscall.Statfs(path, &data)
+	if err != nil {
+		return false
+	}
+
+	fsType := uint32(data.Type)
+
+	if fsType == FsMagicAUFS ||
+		fsType == FsMagicZFS {
```

Comment:
Minor: if we rebase this PR once again, please squeeze this check into a single line.